anonlink/benchmark.py

```diff
@@ -60,70 +64,18 @@ def compute_comparison_speed(n1, n2, threshold):
 
     print("{:6d} | {:6d} | {:4d}e6  ({:5.2f}%) | {:6.3f}  ({:4.1f}% / {:4.1f}%) |  {:8.3f}".format(
         n1, n2, n1*n2 // 1000000,
-        100.0*len(sparse_matrix)/(n1*n2),
+        100.0*len(sparse_matrix[0])/(n1*n2),
         elapsed_time,
         100.0*similarity_time * inv_elapsed_time,
         100.0*solver_time * inv_elapsed_time,
         (n1*n2 / 1e6) * inv_similarity_time))
     return elapsed_time
 
 
-def compare_python_c(ntotal=10000, nsubset=6000, frac=0.8):
-    """Compare results and running time of python and C++ versions.
-
-    :param ntotal: Total number of data points to generate
-    :param nsubset: Number of points for each database
-    :param frac: Fraction of overlap between subsets
-
-    :raises: AssertionError if the results differ
-    :return: dict with 'c' and 'python' keys with values of the total time taken
-             for each implementation
-    """
-
-    nml = NameList(ntotal)
-    sl1, sl2 = nml.generate_subsets(nsubset, frac)
-
-    keys = generate_key_lists('secret', len(nml.schema_types))
-    filters1 = tuple(map(operator.itemgetter(0),
-                         stream_bloom_filters(sl1, keys, nml.SCHEMA)))
-    filters2 = tuple(map(operator.itemgetter(0),
-                         stream_bloom_filters(sl2, keys, nml.SCHEMA)))
-
-    # Pure Python version
-    start = timer()
-    result = anonlink.candidate_generation.find_candidate_pairs(
-        (filters1, filters2),
-        anonlink.similarities.dice_coefficient_python,
-        0.0,
-        k=1)
-    end = timer()
-    python_time = end - start
-
-    # C++ accelerated version
-    start = timer()
-    result3 = anonlink.candidate_generation.find_candidate_pairs(
-        (filters1, filters2),
-        anonlink.similarities.dice_coefficient_accelerated,
-        0.0,
-        k=1)
-    end = timer()
-    cffi_time = end - start
-
-    assert result == result3, "Results are different between C++ cffi and Python"
-
-    # Results are the same
-    return {
-        "c": cffi_time,
-        "python": python_time
-    }
-
-
-def benchmark(size, compare):
+def benchmark(size):
 
     print("Anonlink benchmark -- see README for explanation")
     print("------------------------------------------------")
```

Comment:
I was caught out by PyCharm, as it ignored the anonlink wheel that I built and installed into the virtual environment and rather used the python implementations of solver and dice. Maybe we can add a line here like:
```
print(f"using '{anonlink.solving.greedy_solve.__name__}' as solver and "
          f"'{anonlink.similarities.dice_coefficient.__name__}' as similarity metric")
```
or give a warning if we use the python fallback... Let your imagination sprout.